Michael Brady - Unit 2 - Sprint 4

mbrady4 opened the pull request.

mbrady4 committed:
Initial draft of assignment solution

mbrady4 committed:
General Updates

mbrady4 committed:
Updated Assignment

veritaem reviewed (commented):
because the train and test and so different, and the train score is so low (i would expect in the 250-300 range), and the test is so high (would expect 250-300 again avg) i suspect you may be overfitting your model.  Have you tried making a more general model, like having a boolean flag for some variables (similar to a threshold)?

mbrady4 committed:
General Updates

mbrady4 committed:
Survival Analysis Assignment Complete

mbrady4 committed:
Updated

mbrady4 committed:
Added Lecture notebooks

mbrady4 committed:
Replicated Lecture Notebooks

veritaem reviewed (commented):
judging from the keyboard interrupt on the commit, id imagine the polynomial features weighed it down?  Its a computationally expensive process, and you want to pairplot to see what really needs to be polynomialized

mbrady4 committed:
Assignment Complete

veritaem reviewed (commented):
Your pdp plots and things look very nice but If I may quibble with a couple things, specifically your 'model.fit' and your train test split?
* because the data in the df is ordered by date, and you just split it without shuffling, you lose a great deal of time series info (are more ppl buying bikes? Is pollution going up?).  train_test_split has a default shuffle, it may help?
* because you are estimating the best scoring model on your _training_ data, the best score is just the one that fits all of your already labeled points (which is why its max depth/max trees), which will lead to some overfitting and a lesser prediction score.  Have you tried making the model with a train score and predicting the val score to get your best model out that way?

mbrady4 committed:
Lecture Notebooks

mbrady4 committed:
Built Multivariate Gradient Descent w/Numpy

mbrady4 committed:
Updated Multivariate Gradient Descent Notebook

mbrady4 committed:
Tweaked Feature Selection Notebook

mbrady4 committed:
Updated Pedestrian Crossing Model w/ MAE of 243 on test data

Note, model scored considerably better on the final test data than during cross validation training.

mbrady4 committed:
Initial Sprint Update

Not Final

mbrady4 committed:
Final Sprint Challenge Submission

mbrady4 committed:
Updated Sprint Challenge

Added test ROC_AUC score
